src/pg_query_is_utility_stmt.c

```diff
@@ -0,0 +1,200 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdlib.h>
+
+#include "pg_query.h"
+#include "pg_query_internal.h"
+#include <nodes/nodes.h>
+#include <nodes/nodeFuncs.h>
+
+typedef struct {
+	List *result;
+} UtilityWalkState;
+
+static
+bool
+pg_query_is_utility_stmt_walk(Node *node, UtilityWalkState *state)
```

Comment:
```suggestion
pg_query_is_utility_stmt_walk_actual(RawStmt *raw_stmt)
```

+ don't descend into child nodes (only check the top-level `raw_stmt->stmt` type)